src/agent/onefuzz/src/input_tester.rs

```diff
@@ -47,26 +47,45 @@ impl<'a> Tester<'a> {
         exe_path: &'a Path,
         arguments: &'a [String],
         environ: &'a HashMap<String, String>,
-        timeout: &'a Option<u64>,
-        check_asan_log: bool,
-        check_asan_stderr: bool,
-        check_debugger: bool,
-        check_retry_count: u64,
     ) -> Self {
-        let timeout = Duration::from_secs(timeout.unwrap_or(DEFAULT_TIMEOUT_SECS));
         Self {
             setup_dir,
             exe_path,
             arguments,
             environ,
-            timeout,
-            check_asan_log,
-            check_asan_stderr,
-            check_debugger,
-            check_retry_count,
+            timeout: DEFAULT_TIMEOUT,
+            check_asan_log: false,
+            check_asan_stderr: false,
+            check_debugger: true,
+            check_retry_count: 0,
         }
     }
 
+    pub fn timeout(&mut self, value: u64) -> &mut Self {
```

Comment:
apologies for the late review but was this fixing a clippy issue or just a style change ?

Reply:
@chkeita, my guess is that we were hitting the [`too_many_arguments`](https://rust-lang.github.io/rust-clippy/master/#too_many_arguments) lint, and @bmc-msft just used it a cue to try to improve the overall API.

Reply:
Fair enough, I would like to propose that we keep the type immutable in this case.
Something like 

``` rust
pub fn timeout(self, value: u64) -> Self {
    Self{
        timeout : Duration::from_secs(value),
        ..self
    }
}
```

My reasoning for avoiding mutability is to make it easier to reason about the state of variables because the state can  only be defined at initialization.

Reply:
@ranweiler is correct here. 

I modeled the code after our existing mechanism that uses the builder paradigm, Expand (the variable expansion functionality).  I'm open to switching to the immutable version, but we should probably update both to follow the same paradigm.

Reply:
created #500 with the proposed changed